testplan/common/report/base.py

```diff
@@ -235,6 +235,41 @@ def get_by_uid(self, uid):
         """
         return self._index[uid]
 
+    def __getitem__(self, uid):
+        """Shortcut to `get_by_uid()` method via [] operator."""
+        return self.get_by_uid(uid)
+
+    def set_by_uid(self, uid, item):
+        """
+        Set child report via `uid` lookup.
+
+        Unfortunately, for updating an existing entry requires an O(n) lookup
+        in the list of existing entries, since the index is read-only.
+        Inserting a new entry is O(1).
+
+        :param uid: `uid` for the child report.
+        :type uid: ``hashable``
+        :param item: entry to update or insert into the report.
+        :type item: ``Report``
+        """
+        if uid in self._index:
```

Comment:
Maybe we can save the position of the item in _index instead? In that way we can access the entry as self.entries[self._index[uid]], O(1)

Reply:
Maybe that would be better yeah. TBH storing entries in a list is a mistake, they should be stored in a dict keyed by UID. But I'm wary to make big changes in this area because I know that most of the rest of testplan depends on it, so I'm just making the minimal changes I can.

Reply:
I guess that self._index is "private" to the Report object so changing it to store index should be a safe change, I'll take a look. Good suggestion!